.github/workflows/build-storybook-main.yml

```diff
@@ -0,0 +1,55 @@
+name: Build Storybook Main
+
+on:
+  push:
+    branches:
+      - main
+  workflow_call:
+    secrets:
+      PUBLISH_STORYBOOK_TOKEN:
+        required: true
+
+jobs:
+  build:
+    name: Build Storybook
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: apps/storybook-react
+    steps:
+      - name: Checkout and setup environment
+        uses: MetaMask/action-checkout-and-setup@v1
+        with:
+          is-high-risk-environment: false
+          skip-allow-scripts: true
+
+      - name: Build required packages
+        working-directory: .
+        run: |
+          yarn build
+
+      - name: Build Storybook
+        run: yarn build-storybook
+
+      - name: Upload artifact
+        id: upload
+        uses: actions/upload-pages-artifact@v3
+        with:
+          name: github-pages
+          path: apps/storybook-react/storybook-static
+          retention-days: 1
```

Comment:
Uses the official github action example [upload-pages-artifact](https://github.com/actions/upload-pages-artifact?tab=readme-ov-file)